### PR TITLE
Tests, I got the code to pass all the tests!!!!

### PR DESCRIPTION
**Pull request by colemanliyah**
Thank you Devon!

### CODE REVIEW

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 1]
```diff
@@ -15,9 +15,64 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.stream.Collectors;
```
Comment (RegExrTech) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: I'm not sure the HashSet, Set, List, Arrays, and Collectors imports are used so you could probably remove them. Make sure you rerun the tests after you remove them though to ensure that there won't be any issues!

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 2]
```diff
@@ -15,9 +15,64 @@
...
+    if(occupiedTimes.size()==0) {
+        validTimes.add(whole_day);
+        return validTimes;
+    }
+
+    Collections.sort(occupiedTimes, TimeRange.ORDER_BY_START);
+
+    TimeRange marker = TimeRange.START;
+
+    for(TimeRange meeting : occupiedTimes){
```
Comment (RegExrTech) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: the variable "meeting" is sort of misnamed (I expect this to be an event type object). Something like "meetingTime" or "existingMeetingTime" might be more appropriate.

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 3]
```diff
@@ -15,9 +15,64 @@
...
+        if(meeting.start() > marker.end() && meeting.start()-marker.end() >= request.getDuration()) {
+            validTimes.add(TimeRange.fromStartEnd(marker.end(), meeting.start()-1, true));
+        }
+        if(marker.end() < meeting.end()) {
+            marker = meeting;
+        }
+    }
+    if(marker.end()!=whole_day.end() && whole_day.end()-marker.end()>=request.getDuration()){
```
Comment (RegExrTech) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:69]: Might want to add a comment above this if statement, explaining to the reader why we need to do this last check outside of the loop

**Review by RegExrTech — CHANGES_REQUESTED** [review]
Nicely done! Just a few comments about stuff that doesn't impact the correctness of the solution :)

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java (hunk 3 above)
Comment (colemanliyah) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:69]: Should I be putting in more comments for the other parts of my code?

**Review by RegExrTech — APPROVED** [review]

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java (hunk 3 above)
Comment (RegExrTech) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:69]: No, you should be fine! Commenting is an interesting thing. Typically, for in line comments like this, we only want to do it if it adds something extra that's not obvious from reading the code. Good variable naming and good structure typically mean fewer comments. But sometimes we need to do something and it isn't instantly obvious as to why (or maybe a line of code is very complicated to read). In these cases, having some plain english to describe the code is helpful.

**Pull request merged**